docs/naming-conventions.md

```diff
@@ -0,0 +1,44 @@
+# Naming Conventions
+
+All names in this project should adhere to the guidelines outlined in this document.
+
+## Rust Code
+
+All variable names, function names, module names, and macros in Rust code should follow typical `snake_case` conventions. All Rust types, traits, structs, and enum variants must follow `UpperCamelCase`. Additionally, static and constant variables should be written in `SCREAMING_SNAKE_CASE`. For the complete list of Rust naming conventions, see the [Rust Style Guide](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html).
+
+As a final rule, the functions of all component `ffi/src/lib.rs` files should also follow *snake_case*, but with an additional prefix based on the library of that function. 
+
+Examples:
```

Comment:
If you're going to have examples, show examples of the different uses. 

```
a_function()
MyStruct
A_CONSTANT
```

I like that you found examples from the codebase, so do that.